src/App.js

```diff
@@ -1,26 +1,37 @@
-import React, { Component } from 'react';
+import React, { useState } from 'react';
 import './App.scss';
 import { MoviesList } from './components/MoviesList';
 import { FindMovie } from './components/FindMovie';
-import data from './api/movies.json';
 
-export class App extends Component {
-  state = {
-    movies: data,
-  };
+export const App = () => {
+  const [movies, setMovies] = useState([]);
+  const [error, setError] = useState(false);
+
+  const addNewMovie = (newMovie) => {
+    if (movies.some(movie => movie.imdbID === newMovie.imdbID)
+      || error) {
+      return;
+    }
 
-  render() {
-    const { movies } = this.state;
+    setMovies(current => [...current, newMovie]);
```

Comment:
```suggestion
    setMovies([...movies, newMovie]);
```
Here we can use setMovies without a callback